### PR TITLE
Core: Add list/item group exclusive methods to CollectionState

### PR DESCRIPTION
**Pull request by TRPG0**
It was brought to my attention recently that the way I've been using `has_group` and `count_group` can break completely if the player is to use either Plando or their Starting Inventory to create more instances of an item than there are intended to ever be in a multiworld.

These new methods would help to fix this issue by only adding to the `found` total if any number of the item is in state, before moving on to the next item in the group.

Tested this in Bomb Rush Cyberfunk #2925 by choosing one graffiti design of each size, and adding 20 of them to my starting inventory.

### CODE REVIEW

**Review by ScipioWright — APPROVED** [review]
A very simple addition that is well-needed for some very specific situations.

**Review by Exempt-Medic — APPROVED** [review]
A helpful addition, merged into `main` and tested with a modified Blasphemous world

File: BaseClasses.py [hunk 1]
```diff
@@ -767,13 +767,33 @@ def has_group(self, item_name_group: str, player: int, count: int = 1) -> bool:
             if found >= count:
                 return True
         return False
+
```
Comment (beauxq) [inline BaseClasses.py:770]:
```suggestion

```
white space

File: BaseClasses.py [hunk 2]
```diff
@@ -767,13 +767,33 @@ def has_group(self, item_name_group: str, player: int, count: int = 1) -> bool:
...
+                return True
+        return False
 
     def count_group(self, item_name_group: str, player: int) -> int:
         found: int = 0
         player_prog_items = self.prog_items[player]
         for item_name in self.multiworld.worlds[player].item_name_groups[item_name_group]:
             found += player_prog_items[item_name]
         return found
+
```
Comment (beauxq) [inline BaseClasses.py:789]:
```suggestion

```
removing white space

**Comment by NewSoupVi** [comment]
Hi! Okay, so, I've been holding off on reviewing this because it had a "soft conflict" with https://github.com/ArchipelagoMW/Archipelago/pull/2934, which I wanted to have merged first

Not really a conflict per se, but moreso, now that that PR is in, this implies that this PR should also add `has_from_list_exclusive` and `count_from_list_exclusive`. Can I convince you to add these to the scope of this PR? Then, I'll be willing to review and merge it :)

**Review by NewSoupVi — APPROVED** [review]
Code is correct

+ I timed some of these lines, and it looks like the fastest method was chosen for everything (bool addition is faster than conditionally adding one, `> 0` is faster than bool conversion)

**Pull request merged**